benchmark/conftest.py

```diff
@@ -31,6 +34,15 @@ def delete_csv_files():
             f.write("")
 
 
+def pytest_generate_tests(metafunc):
```

Comment:
do we want (need) to run this every time the fixture is requested, or would it be possible to define this on session or module scope?

Reply:
After having a look at the [pytest docs](https://docs.pytest.org/en/latest/how-to/parametrize.html#pytest-generate-tests) and [this article](https://pytest-with-eric.com/introduction/pytest-generate-tests/):

This is expected behavior and pytest is designed to behave like this. The expected behavior is that pytest calls the `pytest_generate_tests` once for every test in the module in the initial test collection. If the test contains a fixture from the `pytest_generate_tests` function, the test is parameterized. Thus the number of test cases = number how often `pytest_generate_tests` is called to check if there is something to parametrize.

Reply:
But if the fixture content does not change, this could be just run once? I thought that's what fixture scopes are for...

Reply:
In other words, if you separate dataset loading (only once, since the dataset will not change) and function parameterisation (for each test), would that not make it more efficient?

Reply:
Yes, I see the point. I am trying to figure out how to configure this. Then the dataset itself would be a fixture and is loaded once and passed to the `pytest_generate_tests` (in theory - let's see how to implement this :)

Reply:
Fixed it.